src/app/(app)/(modals)/withdraw.tsx

```diff
@@ -0,0 +1,66 @@
+import ArrowLeftIcon from "@assets/icons/arrow-left.svg";
+import { Stack, router } from "expo-router";
+import { useRef, useState } from "react";
+import { Pressable, StyleSheet } from "react-native";
+import PagerView from "react-native-pager-view";
+
+import SelectCurrency from "@/screens/balanceWithdrawal/SelectCurrency";
+import WithdrawAddress from "@/screens/balanceWithdrawal/WithdrawAddress";
+import WithdrawAmount from "@/screens/balanceWithdrawal/WithdrawAmount";
+import WithdrawMethod from "@/screens/balanceWithdrawal/WithdrawMethod";
+
+export default function Withdraw() {
+  const [currentScreen, setCurrentScreen] = useState(0);
+  const ref = useRef<PagerView>(null);
+
+  const setPage = (page: number) => ref.current?.setPage(page);
+
+  function onHeaderLeft() {
+    if (currentScreen !== 0) {
+      setPage(currentScreen - 1);
```

Comment:
Might be silly, but I'd rather use the callback whenever I want to derive a value from the current state.
```suggestion
      setPage(prev => prev - 1);
```
or something like that, just in case there is a race condition, but sure, it's a pagination and probably is not needed.